### PR TITLE
JAMES-2290 Remove io.vavr dependency

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/testing/src/main/java/org/apache/james/utils/DiscreteDistribution.java [hunk 1]
```diff
@@ -73,7 +72,8 @@ private DiscreteDistribution(List<DistributionEntry<T>> distribution) {
     }
 
     public Stream<T> generateRandomStream() {
-        return Stream.continually(this::sample);
+        return Stream.iterate(this, i -> i)
```
Comment (mbaechler) [inline server/testing/src/main/java/org/apache/james/utils/DiscreteDistribution.java]: I was talking about `Stream.generate()` that better fits your need

**Comment by mbaechler** [comment]
I can't really agree with this PR ):

**Comment by chibenwa** [comment]
Is 🍏

**Comment by chibenwa** [comment]
Merged

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**